app-android/app/src/main/java/de/tutao/tutanota/AndroidMobileContactsFacade.kt

```diff
@@ -345,7 +344,8 @@ class AndroidMobileContactsFacade(private val activity: MainActivity) : MobileCo
 	): Boolean {
 		// isNullOrEmpty check is required since department field is a nullableString, comparing the empty string with
 		// null value would fail.
-		return (storedContact.department != serverContact.department && !storedContact.department.isNullOrEmpty())
+		val bothAreNullOrEmpty = storedContact.department.isNullOrEmpty() && serverContact.department.isNullOrEmpty()
+		return (storedContact.department != serverContact.department && !bothAreNullOrEmpty)
```

Comment:
I might be wrong but would the check below also work?

```suggestion
		return ((storedContact.department ?: "") != (serverContact.department ?: ""))
```